src/test/java/HttpUrlConnectionTest.java

```diff
@@ -534,4 +535,33 @@ public void testReplayHttpError() throws Exception {
         Assert.assertNotNull(clientAfterRequest);
         Assert.assertNotNull(clientAfterRequest.getErrorStream());
     }
+
+    @Test
+    public void testCachedInteractionDoesNotExist() throws Exception {
+        Cassette cassette = TestUtils.getCassette("test_cached_interaction_does_not_exist");
+        cassette.erase(); // Erase cassette before recording
+
+        final String url = "https://google.com/path/to/endpoint";
+
+        // make connection using Mode.Record
+        RecordableHttpsURLConnection connection =
+                (RecordableHttpsURLConnection) HttpClients.newClient(HttpClientType.HttpsUrlConnection,
+                        url, cassette, Mode.Record);
+        // make HTTP call (record to cassette)
+        connection.connect();
+        Assert.assertTrue(cassette.numInteractions() > 0); // make sure we recorded something
+
+        // Attempt to replay a cached interaction that does not exist
+        // need to use strict matching to ensure we don't match a different interaction
+        AdvancedSettings advancedSettings = new AdvancedSettings();
+        advancedSettings.matchRules = MatchRules.strict();
+
+        connection = (RecordableHttpsURLConnection) HttpClients.newClient(HttpClientType.HttpsUrlConnection,
+                url + "1", cassette, Mode.Replay, advancedSettings);
+        // make HTTP call (attempt replay from cassette)
+        connection.connect();
+        // Attempt to pull data (e.g. body) from the response (via the input stream)
+        // this throws a RuntimeException because of the way the exceptions are coalesced internally
+        Assert.assertThrows(RuntimeException.class, connection::getInputStream);
```

Comment:
question: can we assert the error message here? If so, let's please do that so we have strong assertions.